packages/core/admin/admin/src/content-manager/components/BlocksEditor/hooks/useBlocksStore.js

```diff
@@ -0,0 +1,320 @@
+    max-width: 100%;
+    padding: ${({ theme }) => theme.spaces[2]};
+  }
+`;
+
+const Blockquote = styled.blockquote.attrs({ role: 'blockquote' })`
+  margin: ${({ theme }) => `${theme.spaces[6]} 0`};
+  font-weight: ${({ theme }) => theme.fontWeights.regular};
+  border-left: ${({ theme }) => `${theme.spaces[1]} solid ${theme.colors.neutral150}`};
+  font-style: italic;
+  padding: ${({ theme }) => theme.spaces[2]} ${({ theme }) => theme.spaces[5]};
+`;
+
+const listStyle = css`
+  margin-block-start: ${({ theme }) => theme.spaces[4]};
+  margin-block-end: ${({ theme }) => theme.spaces[4]};
+  margin-inline-start: ${({ theme }) => theme.spaces[0]};
+  margin-inline-end: ${({ theme }) => theme.spaces[0]};
+  padding-inline-start: ${({ theme }) => theme.spaces[4]};
+
+  ol,
+  ul {
+    margin-block-start: ${({ theme }) => theme.spaces[0]};
+    margin-block-end: ${({ theme }) => theme.spaces[0]};
+  }
+`;
+
+const Orderedlist = styled.ol`
+  list-style-type: decimal;
+  ${listStyle}
+`;
+
+const Unorderedlist = styled.ul`
+  list-style-type: disc;
+  ${listStyle}
+`;
+
+const List = ({ attributes, children, element }) => {
+  if (element.format === 'ordered') return <Orderedlist {...attributes}>{children}</Orderedlist>;
+
+  return <Unorderedlist {...attributes}>{children}</Unorderedlist>;
+};
+
+List.propTypes = {
+  attributes: PropTypes.object.isRequired,
+  children: PropTypes.node.isRequired,
+  element: PropTypes.shape({
+    format: PropTypes.string.isRequired,
+  }).isRequired,
+};
+
+const Img = styled.img`
+  max-width: 100%;
+`;
+
+const Image = ({ attributes, children, element }) => {
+  const { url, alternativeText, width, height } = element.image;
+
+  return (
+    <Box {...attributes}>
+      {children}
+      <Box contentEditable={false}>
+        <Img src={url} alt={alternativeText} width={width} height={height} />
+      </Box>
+    </Box>
+  );
+};
+
+Image.propTypes = {
+  attributes: PropTypes.object.isRequired,
+  children: PropTypes.node.isRequired,
+  element: PropTypes.shape({
+    image: PropTypes.shape({
+      url: PropTypes.string.isRequired,
+      alternativeText: PropTypes.string,
+      width: PropTypes.number,
+      height: PropTypes.number,
+    }).isRequired,
+  }).isRequired,
+};
+
+/**
+ * Manages a store of all the available blocks.
+ *
+ * @returns {{
+ *   [key: string]: {
+ *     renderElement: (props: Object) => JSX.Element,
+ *     variants: {
+ *       [key: string]: {
+ *         icon: React.ComponentType,
+ *         label: {id: string, defaultMessage: string},
+ *         value: Object,
+ *         matchNode: (node: Object) => boolean,
+ *       }
+ *     }
+ *   }
+ * }} an object containing rendering functions and metadata for different blocks, indexed by name.
+ */
+export function useBlocksStore() {
+  return {
+    paragraph: {
+      renderElement: (props) => (
+        <Typography as="p" variant="omega" {...props.attributes}>
+          {props.children}
+        </Typography>
+      ),
```

Comment:
I'm in two minds about this. On the one hand it is very convenient and keeps everything about the block in one place. 

On the other hand this looks similar to the `cellFormatter` pattern and in my opinion that kind of thing becomes difficult to debug.

I feel like the store describes what a block is _and_ how to render it. I'm really not sure if it should or should not do both of those things, but I'm feeling a bit apprehensive about this being here.

Reply:
This is really interesting, I think it would be good to explore how this could be different.